src/Groups/sub.jl

```diff
@@ -850,7 +858,7 @@ julia> typeof(quo(PermGroup, G, N)[1])
 PermGroup
 ```
 """
-function quo(G::T, N::T) where T <: GAPGroup
+function quo(G::T1, N::T2) where {T1 <: GAPGroup, T2 <: GAPGroup}
```

Comment:
We could add a helper
```julia
_is_subgroup_type(::Type{T1}, ::Type{T2}) where {T1, T2} = false
_is_subgroup_type(::Type{T}, ::Type{T}) where {T <: GAPGroup} = true
_is_subgroup_type(::Type{PcGroup}, ::Type{SubPcGroup}) = true
```
and then add checks like
```julia
  @req _is_subgroup_type(T1, T2) "argument types do not fit"
```

Reply:
I thought about introducing something like `_compatibility_check(G1::T1, G2::T2)` which throws an exception if the arguments do not fit together, and otherwise has no effect. Note that checking the types alone is not sufficient, for example in the case of two f.p. subgroups or two pc subgroups which belong to different full groups. If we do not check this compatibility before delegating to GAP then we get just a cryptic error message from GAP.